### PR TITLE
YIM 2022 Email templates

### PR DESCRIPTION
**Pull request by amCap1712**

### CODE REVIEW

File: listenbrainz/webserver/templates/emails/year_in_music.txt [hunk 1]
```diff
@@ -1,29 +1,8 @@
-Dear {{ user_name }},
+{{user_name}}, your ListenBrainz Year in Music 2022 is now available.
 
-The ListenBrainz team would like to wish you happy holidays!
+We have compiled the Year in Music report for 2022 (https://listenbrainz.org/user/{{user_name}}/year-in-music/), which summarises of your listening habits throughout the year. Some of the statistics you can expect to see include your most active listening day of the week, your listening days throughout the year, and users who listened to similar music to you. Additionally, we have included information about the countries where your favorite artists are from.
```
[the remark this answers is not in the document]
Comment (mayhem) [reply listenbrainz/webserver/templates/emails/year_in_music.txt:3]: I think we should nuke the "additionally... "sentence and improve a sentence below:
"To top it off, we've made a map of where your artists are from and we have also generated two new playlists for you to discover even more music in 2023. We hope you enjoy the report and the new playlists!"

**Review by mayhem — CHANGES_REQUESTED** [review]
Looking pretty good. Please make the change to the verbiage and then we should have @reosarevok or @MonkeyDo take a final look at the wording.

File: listenbrainz/webserver/templates/emails/year_in_music.html [hunk 2]
```diff
@@ -1,71 +1,55 @@
...
-    <p>
-        P.S. Heads up! This page loads many album covers; if you are on a mobile device you might want to wait
-        until you are connected to wifi.
-    </p>
-</div>
+        Wishing you a wonderful 2023, from the ListenBrainz team.
+        </p>
+      </div>
+      <div style="padding:0.5em;background-color: #f2f2f2;color: rgba(30, 30, 30, 1);font-size: 9px;text-align: center;font-weight: 300">
+            Whatever small print we usually have at the bottom of our emails, links and junk
```
Comment (MonkeyDo) [inline listenbrainz/webserver/templates/emails/year_in_music.html:51]: Here this text is only a placeholder. Do we have links and stuff we want to put in there? (go to account/read in browser/opt-out from email, etc)
If so let's put those in. If not let's remove the whole div

**Review by MonkeyDo — COMMENTED** [review]
I'll push fixes to the CSS to this branch one I've got it sorted :)

File: listenbrainz/webserver/templates/emails/year_in_music.html (hunk 2 above)
Comment (mayhem) [reply listenbrainz/webserver/templates/emails/year_in_music.html:51]: We don't really have an opt-out at the moment, but I thinks to LB, MB and MeB could be appropriate here, to give the user links to the projects that contributed to this report. But this is not a strong feeling, if we agree to nuke it, then lets do that.

File: listenbrainz/db/year_in_music.py [hunk 3]
```diff
@@ -273,21 +273,23 @@ def send_mail(subject, to_name, to_email, text, html, lb_logo, lb_logo_cid):
...
-              JOIN "user"
-                ON "user".id = yim.user_id
-             WHERE year = :year   
-        """), {"year": year})
-        rows = result.mappings().fetchall()
+    logo_cid = make_msgid()
+    with open("/static/img/year-in-music-22/yim-22-logo-small-compressed.png", "rb") as img:
+        logo = img.read()
+
+    # with db.engine.connect() as connection:
```
[the remark this answers is not in the document]
Comment (mayhem) [reply listenbrainz/db/year_in_music.py:280]: K, just making sure we didn't miss that. :)

File: listenbrainz/db/year_in_music.py [hunk 4]
```diff
@@ -298,31 +300,24 @@ def notify_yim_users(year):
...
-                lb_logo=lb_logo
+                logo_cid=logo_cid,
+                logo=logo
             )
         except Exception:
             current_app.logger.error("Could not send YIM email to %s", user_name, exc_info=True)
 
         # create timeline event too
-        timeline_message = 'ListenBrainz\' very own retrospective on 2021 has just dropped: Check out ' \
+        timeline_message = f'ListenBrainz\' very own retrospective on {year} has just dropped: Check out ' \
```
Comment (reosarevok) [inline listenbrainz/db/year_in_music.py:320]: Shouldn't it be `ListenBrainz\'s`? Or is the idea that the z is pronounced as an s?

File: listenbrainz/webserver/templates/emails/year_in_music.html [hunk 5]
```diff
@@ -1,71 +1,55 @@
...
+      </div>
+      <header style="padding: 2em 1em;background-color: #ff0e25; color:#ffffff;text-align: center;">
+        <p>
+          <span style="text-transform: capitalize;">{{user_name}}</span>, your ListenBrainz Year in Music 2022 is now available.
+        </p>
+        <a href="https://listenbrainz.org/user/{{user_name}}/year-in-music/" style="font-weight: 700;margin-top:1em;color:#ffffff">Click here</a>
+      </header>
+      <div style="padding:1em;color: rgba(30, 30, 30, 1);width: auto;height: auto;font-size: 12px;text-align: center;line-height: normal;">
+        <p>
+          We have compiled the Year in Music report for 2022, which summarises of your listening habits throughout the year.
```
Comment (reosarevok) [inline listenbrainz/webserver/templates/emails/year_in_music.html:37]: summarises of your -> summarises your

File: listenbrainz/webserver/templates/emails/year_in_music.html [hunk 6]
```diff
@@ -1,71 +1,55 @@
...
+          Some of the statistics you can expect to see include your most active listening day of the week, your listening days throughout the year, and users who listened to similar music to you. Additionally, we have included information about the countries where your favorite artists are from.
+        </p>
+        <p>
+        To top it off, we have also generated two new playlists for you to discover even more music in 2023.
+        We hope you enjoy the report and the new playlists.
+        <p>
```
Comment (reosarevok) [inline listenbrainz/webserver/templates/emails/year_in_music.html:43]: `</p>`?

File: listenbrainz/db/year_in_music.py (hunk 4 above)
Comment (mayhem) [reply listenbrainz/db/year_in_music.py:320]: Yes, that is how I've always used it. A little bit playful if you ask me.

File: listenbrainz/webserver/templates/emails/year_in_music.html (hunk 2 above)
Comment (MonkeyDo) [reply listenbrainz/webserver/templates/emails/year_in_music.html]: Nuked. We probably still have a few hours to add it back if we want to.

**Review by mayhem — APPROVED** [review]
Reviewed the text, seems fine now.

**Comment by amCap1712** [comment]
Thanks all!

**Pull request merged**